pkg/actuators/machine/machine.go

```diff
@@ -54,80 +55,250 @@ func newMachineScope(
+			diskBuilder, _ := ovirtC.NewBuildableVMDiskParameters(diskAttachment.DiskID())
+			if ms.machineProviderSpec.Sparse != nil {
+				diskBuilder.MustWithSparse(*ms.machineProviderSpec.Sparse)
+			}
+
+			if ms.machineProviderSpec.Format != "" {
+				diskBuilder.MustWithFormat(ovirtC.ImageFormat(ms.machineProviderSpec.Format))
+			}
+
+			diskParams = append(diskParams, diskBuilder)
+		}
+		optionalVMParams = optionalVMParams.MustWithDisks(diskParams)
+	}
+
+	// Handle Disk Clone
+	if ms.machineProviderSpec.Clone != nil {
+		optionalVMParams = optionalVMParams.MustWithClone(true)
+	} else {
+		if ms.machineProviderSpec.VMType == string(ovirtC.VMTypeDesktop) {
+			optionalVMParams = optionalVMParams.MustWithClone(false)
+		} else {
+			optionalVMParams = optionalVMParams.MustWithClone(true)
+		}
+	}
+
+	instance, err := ms.ovirtClient.CreateVM(ovirtC.ClusterID(clusterId),
+		temp.ID(),
 		ms.machine.Name,
-		ms.machine.Labels["machine.openshift.io/cluster-api-cluster"],
-		ignition,
-		ms.machineProviderSpec)
+		optionalVMParams, ovirtC.ContextStrategy(ms.Context))
+
 	if err != nil {
 		return errors.Wrap(err, "error creating Ovirt instance")
 	}
 
 	// Wait till ready
-	err = ms.waitTillVMReachStatus(ovirtsdk.VMSTATUS_DOWN)
+	_, err = ms.ovirtClient.WaitForVMStatus(instance.ID(), ovirtC.VMStatusDown, ovirtC.ContextStrategy(ms.Context))
 	if err != nil {
 		return errors.Wrap(err, "error creating oVirt VM")
 	}
 
+	var bootableDiskAttachment ovirtC.DiskAttachment
+	newDiskSize := uint64(ms.machineProviderSpec.OSDisk.SizeGB * int64(math.Pow(2, 30)))
+	// Handle OS disk extension
+	if ms.machineProviderSpec.OSDisk != nil {
+		diskAttachments, err := instance.ListDiskAttachments()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to list disk attachments for VM %s.", instance.ID())
+		}
+		for _, diskAttachment := range diskAttachments {
+			if diskAttachment.Bootable() {
+				bootableDiskAttachment = diskAttachment
+			}
+		}
+
+		if bootableDiskAttachment == nil {
+			return errors.Wrapf(err, "VM %s(%s) doesn't have a bootable disk", instance.Name(), instance.ID())
+		}
+
+		disk, err := ms.ovirtClient.GetDisk(bootableDiskAttachment.DiskID(), ovirtC.ContextStrategy(ms.Context))
+		if newDiskSize > disk.ProvisionedSize() {
+			updatedDisk, err := disk.Update(ovirtC.UpdateDiskParams().MustWithProvisionedSize(newDiskSize))
+			if err != nil {
+				return errors.Wrapf(err, "Failed to extend disk %s (%v)", disk.ID())
+			}
+			klog.Infof("Waiting for disk to become OK...")
+			updatedDisk, err = updatedDisk.WaitForOK()
+			if err != nil {
+				return errors.Wrapf(err, "Failed to wait for disk %s to return to OK status. (%v)", disk.ID())
+			}
+		}
+	}
+
+	// handleNics reattachment
+	if ms.machineProviderSpec.NetworkInterfaces != nil && len(ms.machineProviderSpec.NetworkInterfaces) > 0 {
+		nics, err := instance.ListNICs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to list NICs on VM %s", instance.ID())
```

Comment:
We need to check, but I believe that the `ListNICs` already adds this message, so we should be able to just return `err`.